contracts/0.4.24/Lido.sol

```diff
@@ -355,323 +421,301 @@ contract Lido is ILido, StETH, AragonApp {
+    function setMaxPositiveTokenRebase(uint256 _maxTokenPositiveRebase) external {
+        _auth(MANAGE_MAX_POSITIVE_TOKEN_REBASE_ROLE);
+        _setMaxPositiveTokenRebase(_maxTokenPositiveRebase);
+    }
 
-        _setBPValue(FEE_POSITION, _feeBasisPoints);
-        emit FeeSet(_feeBasisPoints);
+    struct OracleReportInputData {
+        // Oracle report timing
+        uint256 timeElapsed;
+        // CL values
+        uint256 clValidators;
+        uint256 clBalance;
+        // EL values
+        uint256 withdrawalVaultBalance;
+        uint256 elRewardsVaultBalance;
+        // Decision about withdrawals processing
+        uint256 requestIdToFinalizeUpTo;
+        uint256 finalizationShareRate;
     }
 
     /**
-    * @notice Set fee distribution
-    * @param _treasuryFeeBasisPoints basis points go to the treasury,
-    * @param _insuranceFeeBasisPoints basis points go to the insurance fund,
-    * @param _operatorsFeeBasisPoints basis points go to node operators.
-    * @dev The sum has to be 10 000.
+    * @notice Updates accounting stats, collects EL rewards and distributes collected rewards if beacon balance increased
+    * @dev periodically called by the Oracle contract
+    * @param _timeElapsed time elapsed since the previous oracle report
+    * @param _clValidators number of Lido validators on Consensus Layer
+    * @param _clBalance sum of all Lido validators' balances on Consensus Layer
+    * @param _withdrawalVaultBalance withdrawal vault balance on Execution Layer for report block
+    * @param _elRewardsVaultBalance elRewards vault balance on Execution Layer for report block
+    * @param _requestIdToFinalizeUpTo right boundary of requestId range if equals 0, no requests should be finalized
+    * @param _finalizationShareRate share rate that should be used for finalization
+    *
+    * @return totalPooledEther amount of ether in the protocol after report
+    * @return totalShares amount of shares in the protocol after report
+    * @return withdrawals withdrawn from the withdrawals vault
+    * @return elRewards withdrawn from the execution layer rewards vault
     */
-    function setFeeDistribution(
-        uint16 _treasuryFeeBasisPoints,
-        uint16 _insuranceFeeBasisPoints,
-        uint16 _operatorsFeeBasisPoints
-    )
-        external
-    {
-        _auth(MANAGE_FEE);
-
-        require(
-            TOTAL_BASIS_POINTS == uint256(_treasuryFeeBasisPoints)
-            .add(uint256(_insuranceFeeBasisPoints))
-            .add(uint256(_operatorsFeeBasisPoints)),
-            "FEES_DONT_ADD_UP"
+    function handleOracleReport(
+        // Oracle timing
+        uint256 _timeElapsed,
+        // CL values
+        uint256 _clValidators,
+        uint256 _clBalance,
+        // EL values
+        uint256 _withdrawalVaultBalance,
+        uint256 _elRewardsVaultBalance,
+        // Decision about withdrawals processing
+        uint256 _requestIdToFinalizeUpTo,
+        uint256 _finalizationShareRate
```

Comment:
Looks like it's not really a `_finalizationShareRate` in the oracle context. It's more of a historical EL value like vault balances. 

Maybe rename and move up?

Reply:
?

Reply:
`_finalizationShareRate` -> `_finalizationShareRateSimulated`

Reply:
done